### PR TITLE
Week 4: Weather App - Alma

### PR DESCRIPTION
**Pull request by almaherris**
## Netlify link
https://clear-skies-ahead.netlify.app/

## Collaborators
[almaherris]

### CODE REVIEW

File: code/script.js [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+////////// API URL storing //////////
+const BASE_URL =
+  "https://api.openweathermap.org/data/2.5/weather?units=metric&q="
+const API_KEY = "206e380881feb4fbbb5f8d99cb75f06d" //query param
+let cityName = "Stockholm" //path param
+const forecastURL =
+  "https://api.openweathermap.org/data/2.5/forecast?units=metric&q="
+
+////////// Function to fetch API data //////////
+const fetchWeatherData = (cityName) => {
```
Comment (HIPPIEKICK) [inline code/script.js:23]: Consider breaking some of the code out from this function to let each function do one thing - i.e. have another function that updates the HTML

File: code/script.js [hunk 2]
```diff
@@ -0,0 +1,286 @@
...
+  fetch(`${BASE_URL}${cityName}&appid=${API_KEY}`)
+    .then((response) => {
+      if (!response.ok) {
+        if (response.status === 404) {
+          // City not found
+          throw new Error("City not found")
+        } else {
+          //Other errors
+          throw new Error("Failed to fetch weather data")
```
Comment (HIPPIEKICK) [inline code/script.js:32]: Nice error handling!

File: code/script.js [hunk 3]
```diff
@@ -0,0 +1,286 @@
...
+
+      //Display in HTML
+      temperature.innerHTML = `${temp}`
+      weather.innerHTML = `${weatherDescriptionCapitalized}`
+      city.innerHTML = cityName
+
+      //Check time to know whether to display day or night image
+      const localTime = new Date(json.dt * 1000) //milliseconds
+      const timeZoneOffset = json.timezone / 3600 //convert to hours
+      localTime.setHours(localTime.getHours() + timeZoneOffset) //adjust time by adding offset
```
Comment (HIPPIEKICK) [inline code/script.js:53]: Nice!

File: code/script.js [hunk 4]
```diff
@@ -0,0 +1,286 @@
...
+        "Dust",
+        "Ash",
+        "Squall",
+        "Tornado",
+      ]
+
+      //Background for daytime 06-20
+      if (formattedLocalTimeInHours >= 6 && formattedLocalTimeInHours < 20) {
+        //If statements changing background based on weather condition
+        if (weatherToday === "Clear") {
```
Comment (HIPPIEKICK) [inline code/script.js:81]: Nice! I like the images you've chosen

File: code/script.js [hunk 5]
```diff
@@ -0,0 +1,286 @@
...
+
+      //Display in HTML adding the formatting created in sunriseOptions
+      sunriseElement.innerHTML = `${sunriseTime}`
+      sunsetElement.innerHTML = `${sunsetTime}`
+    })
+    //Handle errors
+    .catch((error) => {
+      if (error.message === "City not found") {
+        //City not found, only change placeholder text
+        citySearchInput.placeholder = "Try another city"
```
Comment (HIPPIEKICK) [inline code/script.js:142]: Consider changing this back after a city _is_ found

File: code/script.js [hunk 6]
```diff
@@ -0,0 +1,286 @@
...
+      return response.json()
+    })
+    .then((json) => {
+      //Extract today's index
+      const currentDate = new Date()
+      const currentDayIndex = currentDate.getDay()
+
+      //Filter out today's weather data
+      const nextFiveDays = json.list.filter(
+        (element) => new Date(element.dt_txt).getDay() !== currentDayIndex
```
Comment (HIPPIEKICK) [inline code/script.js:184]: ⭐️

File: code/script.js [hunk 7]
```diff
@@ -0,0 +1,286 @@
...
+  if (cityName) {
+    //Fetch weather data and forecast for new city
+    fetchWeatherData(cityName)
+    fetchWeatherForecast(cityName)
+    //Clear the input after user presses enter
+    citySearchInput.value = ""
+  } else {
+    citySearchInput.placeholder = "Try another city"
+  }
+})
```
Comment (HIPPIEKICK) [inline code/script.js:286]: Very nice!

File: code/style.css [hunk 8]
```diff
@@ -0,0 +1,219 @@
...
+}
+
+.app-container {
+  display: flex;
+  flex-direction: column;
+  height: 100vh;
+}
+
+/*Styling for search section*/
+.search-bar {
```
Comment (HIPPIEKICK) [inline code/style.css:20]: It's a little hard to read the place holder and the input text on rainy background

**Review by HIPPIEKICK — APPROVED** [review]
You did a good job implementing the design for the weather app and the code is structured as well. Keep it up 👍

**Review by maikanetaka — COMMENTED** [review]
I am so impressed at your work! You've done a great job organizing everything and making sure it's maintainable. The way you've set up dynamic background changes based on the weather and time of day is super cool! Thanks for let me review your work, I could learn a lot :)

**Pull request closed without merging**